packages/webapp/pages/posts/[id]/index.tsx

```diff
@@ -154,16 +159,28 @@ const PostPage = ({ id, initialData }: Props): ReactElement => {
   ) : (
     <SquadPostPageNavigation squadLink={post.source.permalink} />
   );
-  const articleNavigation = router?.query?.squad ? (
-    <Link href={`/squads/${router.query.squad}`}>
-      <a className="flex flex-row items-center font-bold text-theme-label-tertiary typo-callout">
-        <ArrowIcon size={IconSize.Medium} className="mr-2 -rotate-90" />
-        Back to {router.query.n || 'Squad'}
-      </a>
-    </Link>
-  ) : (
-    <></>
-  );
+
+  const articleNavigation = (() => {
+    const routedFromSquad = router?.query?.squad;
+    const squadLink = `/squads/${router.query.squad}`;
+
+    if (layout === FeedLayout.V1) {
+      return <PostBackButton link={routedFromSquad ? squadLink : undefined} />;
+    }
+
+    if (!routedFromSquad) {
+      return <></>;
```

Comment:
Why do we return an empty fragment here instead of `null`? Is it because of type checks?

Reply:
We can do that, it was because of the way we handle the default value by using `??` which then detects `null` and falls back to the default value but the intention was to actually have the `null` value instead of the fallback.

Just going to have a bit of refactoring, but I am up for it to make it easier to understand overall.